lib/WooCommerce/TransactionalEmails.php

```diff
@@ -61,13 +66,46 @@ public function getEmailHeadings() {
       if (!$settings) {
         $values[$name] = $this->replacePlaceholders($heading['default']);
       } else {
-        $value = isset($settings['heading']) ? $settings['heading'] : $heading['default'];
+        $value = !empty($settings['heading']) ? $settings['heading'] : $heading['default'];
         $values[$name] = $this->replacePlaceholders($value);
       }
     }
     return $values;
   }
 
+  public function useTemplateForWoocommerceEmails() {
+    $this->wp->addAction('woocommerce_init', function() {
+      /** @var callable */
+      $email_header_callback = [\WC()->mailer(), 'email_header'];
+      /** @var callable */
+      $email_footer_callback = [\WC()->mailer(), 'email_footer'];
+      $this->wp->removeAction('woocommerce_email_header', $email_header_callback);
+      $this->wp->removeAction('woocommerce_email_footer', $email_footer_callback);
+      $this->wp->addAction('woocommerce_email_header', function($email_heading) {
+        $this->renderer->render($this->getNewsletter());
+        echo $this->renderer->getHTMLBeforeContent($email_heading);
+      });
+      $this->wp->addAction('woocommerce_email_footer', function() {
+        echo $this->renderer->getHTMLAfterContent();
+      });
+      $this->wp->addAction('woocommerce_email_styles', [$this->renderer, 'prefixCss']);
+    });
+  }
+
+  private function createNewsletter() {
+    $wc_email_settings = $this->getWCEmailSettings();
+    return Newsletter::createOrUpdate([
```

Comment:
Could we use Doctrine for this? We are transitioning towards it but if we keep using the old models we will never finish.

Reply:
This PR is already big, I would make it even bigger by refactoring `Newsletter::createOrUpdate` into Doctrine. What about creating a separate task for that?

Reply:
I don't like this approach. This new task would take a while before it starts. This will go through QA and product QA. 

If we keep doing this we will never completely switch to Doctrine and we've already seen problems with the old ORM. We cannot just use the old models and keep creating tasks for refactoring.

This PR is not THAT big. It seems big because there was a big piece of code moved.